File: programmers/js/level01/176963.js
// 추억 점수
function solution(name, yearning, photo) {
  const nameYearning = name.reduce((acc, cur, idx) => {
    acc[cur] = yearning[idx];
    return acc;
  }, {});

  return photo.map((names) =>
    names.reduce((acc, cur) => acc + (nameYearning[cur] ?? 0), 0)
  );
}

File: programmers/java/level01/176963.java
// 다른 사람의 풀이

import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        List<String> nameList = Arrays.asList(name);

        return Arrays.stream(photo)
            .mapToInt(members -> 
                Arrays.stream(members)
                    .mapToInt(member -> {
                        int index = nameList.indexOf(member);
                        return index == -1 ? 0 : yearning[index];
                    })
                    .sum()
            ).toArray();
    }
}